### PR TITLE
Fix strict mode errors in Name and NameForm scaffold templates

### LINKED ISSUE #6219: [Bug?]: Squiggles in TS strict mode in scaffolded Name.tsx and NameForm.tsx
### What's not working?

Here are some examples

image

image

image

image

image

There are more red squiggles, but I wanted to limit the scope of this issue and my followup PR for a quicker turn-around.

### How do we reproduce the bug?

Just add a prisma model with some fields and run the scaffold generator on that model, while in TS strict mode.

### What's your environment? (If it applies)

_No response_

### Are you interested in working on this?

- [X] I'm interested in working on this

### PR DESCRIPTION
**Pull request by Tobbe**
Fixes #6219 

 - [x] image
 - [x] image
 - [x] image
 - [x] image
 - [x] image

### CODE REVIEW

File: __fixtures__/test-project/package.json [hunk 1]
```diff
@@ -21,8 +21,5 @@
   "prisma": {
     "seed": "yarn rw exec seed"
   },
-  "packageManager": "yarn@3.2.2",
```
Comment (dac09) [inline __fixtures__/test-project/package.json:24]: This needs to be put back for now.... :/

**Comment by Tobbe** [comment]
> There seems to be a lot of unrelated changes here Tobbe?

Yeah. I must have rebuilt the fixture with binaries built on another branch or something. Not totally sure tbh. Cleaned up now though 🙂

**Comment by dac09** [comment]
Looks ok to me @Tobbe - what do you think of aliasing the type just above the prop types though? 

`type ContactProp = NonNullable<...>`

And that way we get to reuse it without having to type that long type everywhere?

**Comment by dac09** [comment]
Haha for once I wouldn't have given it such a long variable name

**Comment by Tobbe** [comment]
Haha. Happy to change if you want, and have a better suggestion 🙂  I didn't like having `Prop` in the name because the type is also used for args (for `onSave`). And I thought just `Contact` was a bit misleading.

**Comment by dac09** [comment]
I think when you're writing the component - it doesn't make sense to hold so much detail about the types in your head.  In a form, you supply a contact and edit a contact - the fact that it's nonnullable doesn't need to be in your mind the whole time, but only when you look at the type definition.

**Comment by dac09** [comment]
(But when you're writing the templates and fixing typescript issues, you do)

**Comment by Tobbe** [comment]
So what's your suggestion? Just `Contact`? Don't you think that's confusing with the `Contact` type we already have in types/graphql? How about `FormContact`?

**Comment by dac09** [comment]
Yeah FormContact makes sense!

**Pull request merged**